planar/intersect/xsweep.go

```diff
@@ -33,29 +33,23 @@ func (et eventType) String() string {
 }
 
 type event struct {
-	edge     int         // the indext number of the edge in the segment list.
-	edgeType eventType   // Is this the left or right edge.
-	ev       *geom.Point // event vertex
+	edge     int        // the indext number of the edge in the segment list.
+	edgeType eventType  // Is this the left or right edge.
+	ev       geom.Point // event vertex
```

Comment:
We can avoid additional allocations by using geom.Point, which is [2]float64 and can be embedded directly in the struct.